### PR TITLE
Bug/ex UI 2431 special characters crashing

### PR DESCRIPTION
**Pull request by connorpgpmcelroy**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EXUI-2431

### Change description ###
Edit special characters before they are encoded to ensure there are no errors with non-ASCII characters

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Review by RiteshHMCTS — DISMISSED** [review]

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]
Let's talk about this at the 3 amigos

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/case-edit-utils/case-edit.utils.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    // note: this could be done for other characters provided they work for btoa()
+    return character;
+  }
+  // Note: Will convert to HTML entity
+  return CaseEditUtils.PREFIX + character.charCodeAt(0) + CaseEditUtils.SUFFIX;
+}
+
+export class CaseEditUtils {
+
+  // Note: Made these purposely odd to avoid possible data conflicts
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/case-edit-utils/case-edit.utils.ts:14]: This comment is now incorrect

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/case-edit-utils/case-edit.utils.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  public static readonly PREFIX = '&#';
+  public static readonly SUFFIX = ';';
+
+  public convertNonASCIICharacters(rawString: string): string {
+    return rawString ? rawString.replace(/[^\x20-\x7F]/g, function (c) {
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/case-edit-utils/case-edit.utils.ts:19]: Sorry to be annoying, but I think that 0x7F is a control character (Delete) so it should be replaced - this should be /[^\x20-\x7E]/

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]
Happy with basic approach, but a couple more minor changes

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/services/cases.service.ts [hunk 3]
```diff
@@ -396,8 +397,12 @@ export class CasesService {
   private addClientContextHeader(headers: HttpHeaders): HttpHeaders {
     const clientContextDetails = this.sessionStorageService.getItem('clientContext');
     if (clientContextDetails) {
-      // may require URI encoding in certain circumstances
-      const clientContext = window.btoa(clientContextDetails);
+      let clientContextEdit = JSON.parse(clientContextDetails);
+      clientContextEdit.client_context.user_task.task_data.name = 'Review ㅪ the ㋚ appeal \`';
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/case-editor/services/cases.service.ts:401]: Is this left over debug code?

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]

**Review by andywilkinshmcts — APPROVED** [review]

**Pull request merged**